functions/Rename-DbaLogin.ps1

```diff
@@ -120,39 +126,43 @@ function Rename-DbaLogin {
             }
 
-            foreach ($db in $dbenums) {
-                $db = $databases[$db.DBName]
-                $user = $db.Users[$Login]
-                Write-Message -Level Verbose -Message "Starting update for $db"
-
-                if ($Pscmdlet.ShouldProcess($SqlInstance, "Changing database $db user $user from [$Login] to [$NewLogin]")) {
-                    try {
-                        $oldname = $user.name
-                        $user.Rename($NewLogin)
-                        [pscustomobject]@{
-                            ComputerName = $server.ComputerName
-                            InstanceName = $server.ServiceName
-                            SqlInstance  = $server.DomainInstanceName
-                            Database     = $db.name
-                            PreviousUser = $oldname
-                            NewUser      = $NewLogin
-                            Status       = "Successful"
-                        }
-
-                    } catch {
-                        Write-Message -Level Warning -Message "Rolling back update to login: $Login"
-                        $currentLogin.rename($Login)
-
-                        [pscustomobject]@{
-                            ComputerName = $server.ComputerName
-                            InstanceName = $server.ServiceName
-                            SqlInstance  = $server.DomainInstanceName
-                            Database     = $db.name
-                            PreviousUser = $NewLogin
-                            NewUser      = $oldname
-                            Status       = "Failure to rename. Rolled back change."
+            if ($Force) {
+                foreach ($db in $dbenums) {
+                    $db = $databases[$db.DBName]
+                    $user = $db.Users[$Login]
+                    if ($user) {
+                        Write-Message -Level Verbose -Message "Starting update for $db"
+
+                        if ($Pscmdlet.ShouldProcess($SqlInstance, "Changing database $db user $user from [$Login] to [$NewLogin]")) {
+                            try {
+                                $oldname = $user.name
+                                $user.Rename($NewLogin)
+                                [pscustomobject]@{
+                                    ComputerName = $server.ComputerName
+                                    InstanceName = $server.ServiceName
+                                    SqlInstance  = $server.DomainInstanceName
+                                    Database     = $db.name
+                                    PreviousUser = $oldname
+                                    NewUser      = $NewLogin
+                                    Status       = "Successful"
+                                }
+
+                            } catch {
+                                Write-Message -Level Warning -Message "Rolling back update to login: $Login"
+                                $currentLogin.rename($Login)
```

Comment:
Do you really want to rollback the login rename? In line 105 there was already an output stating "Successful". So better wait with the output untill all changes are made.

Reply:
agreed, thanks, will do

Reply:
done! plz review

Reply:
don't know if `$currentLogin.rename($Login)` returns something, but in line 140 you used `$null = `. (And it's "Rename" wirh "R".)

Reply:
fixed to make it more uniform!